fixtures/flight/src/index.js

```diff
@@ -2,20 +2,32 @@ import * as React from 'react';
 import {Suspense} from 'react';
 import ReactDOM from 'react-dom';
 import ReactTransportDOMClient from 'react-transport-dom-webpack';
+import {RefreshContext} from './Context.client';
 
-let data = ReactTransportDOMClient.createFromFetch(
+let initialData = ReactTransportDOMClient.createFromFetch(
   fetch('http://localhost:3001')
 );
 
 function Content() {
-  return data.readRoot();
+  let [data, setData] = React.useState(initialData);
+
+  function refresh() {
+    setData(
+      ReactTransportDOMClient.createFromFetch(fetch('http://localhost:3001'))
```

Comment:
Once we have it, this should use the client cache and it's the cache that gets reset.

However, even in that world this can be optimized by seeding the client cache with a new entry.

The idiomatic way would be to seed it with a response that comes from the same HTTP request as the mutation though.

That last part you can prepare for now even without a client cache. You just need to pass this in to the refresh function as an argument.

Reply:
Just to be explicit. The reason for this is not perf but eventual consistency. You can easily get into a state where you never see your mutation in prod because the propagation is always slower than immediately handling the next request.